src/components/button/button.twig

```diff
@@ -1,19 +1,36 @@
-{% set tagName = tagName | default(href ? 'a' : 'button') %}
+{% set tag_name = tag_name | default(href ? 'a' : 'button') %}
 
-<{{tagName}}
+<{{tag_name}}
   class="c-button{% if class %} {{class}}{% endif %}"
-  {% if ariaExpanded %}
-    aria-expanded="{{ariaExpanded}}"
+  {% if aria_expanded %}
+    aria-expanded="{{aria_expanded}}"
   {% endif %}
-  {% if tagName == 'a' %}
+  {% if tag_name == 'a' %}
     href="{{href|default('#')}}"
-  {% elseif tagName == 'button' %}
+  {% elseif tag_name == 'button' %}
     type="{{type|default('button')}}"
     {% if disabled %}
       disabled
     {% endif %}
-  {% endif %}>
+  {% endif %}
+>
+  {% block icon_start %}
+    {% if icon_start %}
+      {% include '@cloudfour/components/icon/icon.twig' with {
+        name: icon_start,
+        inline: true
+      } %}
+    {% endif %}
+  {% endblock %}
   {% block content %}
     {{label | default('Hello world')}}
   {% endblock %}
-</{{tagName}}>
+  {% block icon_end %}
+    {% if icon_end %}
+      {% include '@cloudfour/components/icon/icon.twig' with {
+        name: icon_end,
+        inline: true
+      } %}
+    {% endif %}
+  {% endblock %}
```

Comment:
I opted to name both the block and the argument the same (e.g. `icon_end`).

This allows for the component consumer to use the `icon_end` both as an argument or as a block:
```twig
{% embed '@cloudfour/components/button/button.twig' with { icon_end: 'search' } %}
  {% block content %}Find it!{% endblock %}
{% endembed %}
```
Or:
```twig
{% embed '@cloudfour/components/button/button.twig' %}
  {% block content %}Button with a custom icon{% endblock %}
  {% block icon_end %}
    <svg viewBox="0 0 24 24" width="24" height="24" class="c-icon c-icon--inline">
      <circle r="8" cx="12" cy="12"></circle>
    </svg>
  {% endblock %}
{% endembed %}
```

I keep going back & forth on whether this is confusing or not. If both are used at the same time, the block wins.

Other ideas were to name the block something else (e.g. `icon_end_block`).

How do you all feel about this choice? 🤔

Reply:
I'm open to differing points of view, but this is my take.

If I use the `icon_end` property (_not_ the block), the result _must_ be an icon. There is absolutely no way to make `icon_end` do anything other than display an icon at the end.

But that isn't really true of the block. Even though it's called `icon_end`, there is absolutely nothing stopping me from putting whatever I wanted to there... an image of a kitten, a bit of additional text, a `<span>`. Even though the block is _called_ `icon_end`, it's really just a placeholder for "stuff that follows the content."

So if the block's purpose is more generic than the property's, I'm a fan of naming the property something more specific than the block.

Happy to chat with voices or to consider other points of view on this! That's just my instinct.

Reply:
Excellent point, @tylersticka, thank you for your perspective!

I'll look at renaming the property and/or the block. 👍

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1264/commits/2f712059b3fe0591b94d1cedff2aaf82bcc43ff3